.github/workflows/build-extra.yml

```diff
@@ -2,51 +2,29 @@ name: Build-extra CI
 
 on:
   push:
-    paths-ignore:
-      - '.github/ISSUE_TEMPLATE/*'
-      - 'contrib/syntax/**'
-      - 'contrib/vim/**'
-      - 'etc/**'
-      - 'src/man/*.in'
-      - .git-blame-ignore-revs
-      - .github/dependabot.yml
-      - .github/pull_request_template.md
-      - .github/workflows/build.yml
-      - .github/workflows/codeql-analysis.yml
-      - .github/workflows/codespell.yml
-      - .github/workflows/profile-checks.yml
-      - .gitignore
-      - .gitlab-ci.yml
-      - CONTRIBUTING.md
-      - COPYING
-      - README
-      - README.md
-      - RELNOTES
-      - SECURITY.md
-      - src/firecfg/firecfg.config
+    paths:
+      - 'src/**.c'
+      - 'src/**.h'
+      - 'src/**.mk'
+      - 'src/**Makefile'
+      - .github/workflows/build-extra.yml
+      - Makefile
+      - config.mk.in
+      - config.sh.in
+      - configure
+      - configure.ac
   pull_request:
-    paths-ignore:
-      - '.github/ISSUE_TEMPLATE/*'
-      - 'contrib/syntax/**'
-      - 'contrib/vim/**'
-      - 'etc/**'
-      - 'src/man/*.in'
-      - .git-blame-ignore-revs
-      - .github/dependabot.yml
-      - .github/pull_request_template.md
-      - .github/workflows/build.yml
-      - .github/workflows/codeql-analysis.yml
-      - .github/workflows/codespell.yml
-      - .github/workflows/profile-checks.yml
-      - .gitignore
-      - .gitlab-ci.yml
-      - CONTRIBUTING.md
-      - COPYING
-      - README
-      - README.md
-      - RELNOTES
-      - SECURITY.md
-      - src/firecfg/firecfg.config
+    paths:
+      - 'src/**.c'
+      - 'src/**.h'
+      - 'src/**.mk'
+      - 'src/**Makefile'
+      - .github/workflows/build-extra.yml
+      - Makefile
+      - config.mk.in
+      - config.sh.in
+      - configure
+      - configure.ac
```

Comment:
What about ci, m4 and contrib folders? And `*.sh`? Maybe tests folder?

Reply:
> What about ci, m4 and contrib folders? And `*.sh`? Maybe tests folder?

Good catch, fixed the missing m4/ and added comments to clarify what each
workflow is supposed to do.

ci/ is currently only used for the profile checks, so it's only in
check-profiles.yml.

contrib/, test/ and some shell scripts are used in `make dist`, so they are in
build.yml (which in this PR checks both `make dist` and the normal build).

test/ is otherwise only used when running tests, so it's only in test.yml.

The shell scripts in the root directory do not seem to be used for anything
during the normal build, so they wouldn't affect the checks.

Is there any script in particular that you think would affect it?

Reply:
> ci/ is currently only used for the profile checks, so it's only in
check-profiles.yml.

`./ci/printenv.sh`

Reply:
> > ci/ is currently only used for the profile checks, so it's only in
> > check-profiles.yml.
> 
> `./ci/printenv.sh`

That makes sense, though since this is only used for diagnostics in CI, how
about adding it only to build.yml to avoid re-running everything on every
change?

Reply:
On second thought, this file is unlikely to be changed often anyway, so I added
it to every workflow.